### PR TITLE
Supported CommandNameMapping. See: https://github.com/eventuate-tram/…

### PR DESCRIPTION
**Pull request by dartartem**
…eventuate-tram-core/issues/164

### CODE REVIEW

File: reactive-schema.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+USE eventuate;
```
Comment (cer) [inline reactive-schema.sql:1]: What is reactive about this schema?

File: orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                      .to("customerService")
+                      .build());
+  }
+
+  private Mono<Void> approve(CreateOrderSagaData data) {
+    return Mono.defer(() -> orderService.approveOrder(data.getOrderId()));
+  }
+
+  private Mono<Void> reject(CreateOrderSagaData data) {
+    return Mono.defer(() -> orderService.rejectOrder(data.getOrderId(), data.getRejectionReason()));
```
Comment (cer) [inline orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java:74]: Why defer()?

File: orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/customers/service/CustomerService.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    return possibleCustomer
+            .flatMap(customer -> {
+              return creditReservationRepository
+                      .findAllByCustomerId(customerId)
+                      .collectList()
+                      .flatMap(creditReservations -> handleCreditReservation(creditReservations, customer, orderId, customerId, orderTotal));
+            })
+            .switchIfEmpty(Mono.error(new CustomerNotFoundException()))
+            .as(transactionalOperator::transactional)
```
Comment (cer) [inline orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/customers/service/CustomerService.java:50]: Isn't the transaction managed by the message handler?

**Review by cer — CHANGES_REQUESTED** [review]

File: reactive-schema.sql (hunk 1 above)
Comment (dartartem) [reply reactive-schema.sql:1]: I meant schema for reactive tests. I will update.

File: orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java (hunk 2 above)
Comment (dartartem) [reply orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java:74]: removed

File: orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            })
+            .then();
+  }
+
+  private Mono<CommandWithDestination> reserveCredit(CreateOrderSagaData data) {
+    Long customerId = data.getOrderDetails().getCustomerId();
+    Money orderTotal = data.getOrderDetails().getOrderTotal();
+
+    return Mono
+            .fromSupplier(() -> send(new ReserveCreditCommand(customerId, data.getOrderId(), orderTotal))
```
[the remark this answers is not in the document]
Comment (dartartem) [reply orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java:64]: Fixed

File: orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          .step()
+            .invokeParticipant(this::reserveCredit)
+            .onReply(CustomerNotFound.class, this::handleCustomerNotFound)
+            .onReply(CustomerCreditLimitExceeded.class, this::handleCustomerCreditLimitExceeded)
+          .step()
+            .invokeLocal(this::approve)
+          .build();
+
+  private Mono<Void> handleCustomerNotFound(CreateOrderSagaData data, CustomerNotFound reply) {
+    return Mono.fromRunnable(() -> data.setRejectionReason(RejectionReason.UNKNOWN_CUSTOMER));
```
[the remark this answers is not in the document]
Comment (dartartem) [reply orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java:37]: Fixed

File: orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  private Mono<Void> handleCustomerCreditLimitExceeded(CreateOrderSagaData data, CustomerCreditLimitExceeded reply) {
+    return Mono.fromRunnable(() -> data.setRejectionReason(RejectionReason.INSUFFICIENT_CREDIT));
+  }
+
+  @Override
+  public ReactiveSagaDefinition<CreateOrderSagaData> getSagaDefinition() {
+    return this.sagaDefinition;
+  }
+
+  private Mono<Void> create(CreateOrderSagaData data) {
```
[the remark this answers is not in the document]
Comment (dartartem) [reply orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/orders/service/CreateOrderSaga.java:49]: Fixed

File: orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/customers/service/CustomerService.java (hunk 3 above)
Comment (dartartem) [reply orders-and-customers-spring-reactive/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/spring/reactive/customers/service/CustomerService.java:50]: Yes

**Pull request merged**